### PR TITLE
Move to POST instead of DELETE on notifications unregister API call

### PR DESCRIPTION
**Pull request by BloodyPixy**
Referencing this https://github.com/TrustWallet/trust-wallet-ios/issues/763 ticket.

### CODE REVIEW

File: Trust/Settings/Coordinators/PushNotificationsRegistrar.swift [hunk 1]
```diff
@@ -33,7 +33,20 @@ class PushNotificationsRegistrar {
             preferences: NotificationsViewController.getPreferences()
         )
 
-        trustProvider.request(.unregister(device: device)) { _ in }
+        trustProvider.request(.unregister(device: device)) { response in
+            guard response.error == nil else {
```
Comment (vikmeup) [inline Trust/Settings/Coordinators/PushNotificationsRegistrar.swift]: just call `trustProvider.request(.unregister(device: device))`, no need to handle response

File: Trust/EtherClient/TrustClient/TrustClient.swift [hunk 2]
```diff
@@ -45,7 +45,7 @@ extension TrustService: TargetType {
         case .getTokens: return .get
         case .getTransaction: return .get
         case .register: return .post
-        case .unregister: return .delete
+        case .unregister: return .post
```
Comment (vikmeup) [inline Trust/EtherClient/TrustClient/TrustClient.swift]: @kolya182 what's the reason to use `post` instead of `delete`?

Comment (kolya182) [reply Trust/EtherClient/TrustClient/TrustClient.swift]: @vikmeup Following REST API rules for`delete` end point should looks like `DELETE` push/unregister/{TOKEN} in this case we would expose TOKEN in logs, which is not ideal. So as a trade off we choose `post` which is not ideal either

File: Trust/Settings/Coordinators/PushNotificationsRegistrar.swift (hunk 1 above)
Comment (kolya182) [reply Trust/Settings/Coordinators/PushNotificationsRegistrar.swift]: @vikmeup What if server is down or user has bad internet connection ?. Hi would think subscription went fine, but in really not.

Comment (vikmeup) [reply Trust/Settings/Coordinators/PushNotificationsRegistrar.swift]: it's ok. no need to handle it. Let's assume it always succeed

**Comment by BloodyPixy** [comment]
@vikmeup is this PR going to be approved or closed?

**Comment by vikmeup** [comment]
@BloodyPixy nice work! closed

**Pull request merged**